src/app/StockBase/CollaboratorBase.js

```diff
@@ -29,6 +34,40 @@ class ColaboratorBase extends Base {
       throw { status: 401, message: 'Verifique seu documento e senha' };
     }
   }
+
+  async updateProfile(data, idCollaborator) {
+    const { password, newPassword, newConfirmPassword } = data;
+
+    const collaborator = await super.findOne({ where: { idCollaborator } });
+    if (!collaborator) throw { status: 401, message: 'Usuário não encontrado' };
+
+    const isPasswordValid = await collaborator.verifyPassword(password);
+
+    if (!isPasswordValid)
+      throw { status: 401, message: 'Verifique seu documento e senha' };
+
+    if (newPassword && newConfirmPassword) {
+      if (newPassword !== newConfirmPassword)
+        throw { status: 401, message: 'Senhas não correspondem' };
+      data.password = await hashPass(newPassword);
+    }
+
+    data.password = await hashPass(password);
+    const collaboratorUpdated = await super.update(data, {
+      where: { idCollaborator },
+    });
+    return collaboratorUpdated ? true : false;
+  }
+
+  async updateCollaborator(data, idCollaborator) {
+    const collaborator = await super.findOne({ where: { idCollaborator } });
+    if (!collaborator) throw { status: 401, message: 'Usuário não encontrado' };
```

Comment:
acho que 401 n cabe pra esse caso. Melhor seria 404, se idCollaborator for passado pela rota, se n pode ser 400